Upgrade packages and move to ESLint

benjay10 opened the pull request:
Remove old deprecated TSLint and move to ESLint.

As a newcomer to TypeScript, I'm not familiar with the tools. Trying to run the code from this repository led to package dependency problems, probably caused by newer version of TypeScript conflicting the maximum supported versions of TypeScript compatible with the TSLint system. With this PR I tried to move to ESLint instead with the same, mostly default configuration, update some other packages, and solve some minor linting issues at once.

benjay10 committed:
Upgrade packages and move to eslint

Remove old deprecated tslint and move to eslint.

benjay10 committed:
Revert more changes that belong to other PR

Again, my apologies for this awkward git dancing.

rubensworks commented:
Thanks for this @benjay10! A migration to eslint has indeed been needed for a while.

However, to have some consistency across projects, I always use the same eslint config within my projects.
Would it be possible to copy the eslint setup as used here: https://github.com/rubensworks/rdf-data-factory.js ?

benjay10 committed:
Added rest of config similar to other projects

As asked, the config for this project has now been copied from
https://github.com/rubensworks/rdf-data-factory.js.
This caused a lot of linting issues, to they will be fixed in the next
commit(s).

benjay10 committed:
All linting fixes

I hope this did not break anything. This was not easy to do.

benjay10 commented:
I think this is now more in line with the rest of your projects.

By introducing the new ESLint configuration, there were hundreds of linting problems, of course. I hope nothing has broken. Probably not as all the tests were still correct.
There is also still some problem with package dependency version. Your `@rubensworks/eslint-config` relies on older `@typescript-eslint/eslint-plugin` (and other) packages, so I had to downgrade them here.

rubensworks commented on package.json:
```diff
@@ -59,16 +67,27 @@
     "prepare": "npm run build",
     "version": "manual-git-changelog onversion"
+  },
+  "resolutions": {
```
This can be removed.

benjay10 replied:
Done

rubensworks reviewed (changes requested):
Looks perfect!
Just one minor thing, then this should be good to go 🙂

benjay10 committed:
Removed resolutions from package.json

rubensworks commented:
Thanks @benjay10!
